client/geo.py

```diff
@@ -4,41 +4,66 @@
 import geopy.geocoders
 from django.conf import settings
 
-
-class Locators:
-    NOMINATIM = 'NOMINATIM'
-    GOOGLE = 'GOOGLE'
-    ARCGIS = 3
+# Used to convert city names with added information from MOI to proper city names readable by geocoders.
+CITY_CLEANUP_RE = re.compile(r"^(.*?)(?: \(.*\))?$")
 
 
-CITY_CLEANUP_RE = re.compile(r"^(.*?)(?: \(.*\))?$")
+def run_query(address, coder=None, cache_dict=None, sleep=False):
+    if coder is None:
+        coder = get_geocoder()
+    if cache_dict is not None and address in cache_dict:
+        return cache_dict[address]
+    # Required by some terms of service.
+    if sleep:
+        time.sleep(1)
+    return coder.geocode(address)
 
 
-def get_coordinates(city_name, address):
+def get_coordinates(city_name, address, cache_dict=None):
     city_name = CITY_CLEANUP_RE.match(city_name).group(1)
     coder = get_geocoder()
     built_address = "%s, %s, ישראל" % (address, city_name)
-    location = coder.geocode(built_address)
-    # If coding fails, attempt to resolve city_name.
+    location = run_query(built_address, coder=coder, cache_dict=cache_dict)
+    # If coding fails, attempt to resolve location by city.
     if not location:
-        # Required by Nominatim terms of service.
-        if settings.LOCATOR == Locators.NOMINATIM:
-            time.sleep(1)
         built_address = "%s, ישראל" % city_name
-        location = coder.geocode(built_address)
+        location = run_query(
+            built_address,
+            coder=coder,
+            cache_dict=cache_dict,
+            sleep=(settings.LOCATOR == settings.LOCATOR.nominatim)
+        )
 
         if not location:
             raise LookupError("Failed to convert address to coordinates.")
+    if cache_dict is not None:
+        cache_dict[built_address] = location
     return location
 
 
+def __bulk_geocode(addresses_list: list) -> dict:
+    """
+    WARNING: This function is UNTESTED. USE AT YOUR OWN RISK.
+    :param addresses_list: List of tumples in the structure (city_name, address).
+    :type addresses_list: list(tuple(city_name, address))
+    :returns: A dict of Location objects corresponding to input addresses.
+    """
+    results_dict = {}
+    cache_dict = {}
+    for address_tuple in addresses_list:
+        results_dict[address_tuple] = get_coordinates(address_tuple[0], address_tuple[1], cache_dict=cache_dict)
+        if settings.LOCATOR == settings.LOCATOR.nominatim:
+            time.sleep(1)
+    return results_dict
+
+
 def get_geocoder():
     coder = None
-    if settings.LOCATOR == Locators.NOMINATIM:
+    if settings.LOCATOR == settings.LOCATOR.nominatim:
```

Comment:
Use settings.LocatorTypes.nominatim and not settings.LOCATOR.nominatim

Reply:
Had some access errors, preferred to keep it simple.